main/src/ca/uwaterloo/flix/util/Formatter.scala

```diff
@@ -31,7 +31,7 @@ trait Formatter {
     def leftline: String = {
       val sb = new StringBuilder()
       for (lineNo <- beginLine to endLine) {
-        val currentLine = loc.lineAt(lineNo)
+        val currentLine = loc.lineAt(lineNo).trim.replaceAll(System.lineSeparator(), "")
```

Comment:
Can you try to change lineAt inside `SourceLocation` to replace `\n` by "" and `\r` by "", i.e. two replaces. That should work!